### DISCUSSION TITLE
Clickhouse occupies a lot of operating system memory

### DISCUSSION THREAD

**Discussion by xkz130** [ClickHouse/ClickHouse#48371] [Q&A] [answered]
Deploying the clickhouse service using K8S, it was found that the operating system memory resources were running out, limiting the use of clickhouse memory to 10G. It was found that the maximum use could also reach 10G, and then limited to 20G. The maximum use could still reach 20G. After the memory usage reached its highest point, it sharply dropped to 0, and then rose to its highest point again. This situation repeatedly occurred, with the operating system bit being centros7.6
![image](https://user-images.githubusercontent.com/46994912/229713783-97965369-2df3-4c83-a6c0-ff70b6b6e612.png)
![image](https://user-images.githubusercontent.com/46994912/229713900-67be888d-2202-4445-98b4-6c71951186a1.png)

**Comment by mdonkers** [comment, marked as answer]
Depending on the workloads you are running, ClickHouse may indeed show high memory usage. This is expected though (https://clickhouse.com/docs/en/operations/tips#ram)

Could you explain a bit more what workloads ClickHouse was running, as some tuning might be possible for smaller RAM.

Also, please make sure the `system.query_log` is enabled (https://clickhouse.com/docs/en/operations/server-configuration-parameters/settings#server_configuration_parameters-query-log) and then you could check the `system.query_log` table for the amount of memory used by specific queries (see https://clickhouse.com/docs/en/operations/system-tables/query_log#system_tables-query_log). Also the `system.processes` table might give you more information about what processes occupy memory (see https://clickhouse.com/docs/en/operations/system-tables/processes)
